App is Crashing on Firebase ID issue

**Description:**
<!-- (write below this line) -->
My App with oneSignal was working fine.  Just 4 days ago my app is started crashing on building from android studio on Launch whereas it is working perfectly fine on play store.

Here is the crash I am getting -
```java
java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process Make sure to call FirebaseApp.initializeApp(Context) first.
        at com.google.firebase.FirebaseApp.getInstance(Unknown Source)
        at com.google.firebase.iid.FirebaseInstanceId.getInstance(Unknown Source)
        at com.google.firebase.iid.FirebaseInstanceIdService.zza(Unknown Source)
        at com.google.firebase.iid.FirebaseInstanceIdService.handleIntent(Unknown Source)
```
**Environment**
1. Android SDK 7.0
2. Onesignal dependency in app
com.onesignal:OneSignal:3.+@aar
3. Google Play Services Version
com.google.android.gms:play-services:10.2.6
4.Android Support lib
com.android.support:appcompat-v7:25.2.0

This didn't  worked for me.

Stack trace:

`06-30 09:11:23.334 29748-30190/br.com.jcs.torktransportes.torkapp E/AndroidRuntime: FATAL EXCEPTION: pool-7-thread-1
                                                                                    Process: br.com.jcs.torktransportes.torkapp, PID: 29748
                                                                                    java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process br.com.jcs.torktransportes.torkapp. Make sure to call FirebaseApp.initializeApp(Context) first.
                                                                                        at com.google.firebase.FirebaseApp.getInstance(Unknown Source)
                                                                                        at com.google.firebase.iid.FirebaseInstanceId.getInstance(Unknown Source)
                                                                                        at com.google.firebase.iid.FirebaseInstanceIdService.zza(Unknown Source)
                                                                                        at com.google.firebase.iid.FirebaseInstanceIdService.handleIntent(Unknown Source)
                                                                                        at com.google.firebase.iid.zzb$1.run(Unknown Source)
                                                                                        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
                                                                                        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
                                                                                        at java.lang.Thread.run(Thread.java:761)
06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: Rejecting re-init on previously-failed class java.lang.Class<com.onesignal.OneSignalChromeTab$OneSignalCustomTabsServiceConnection>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/support/customtabs/CustomTabsServiceConnection;
06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art:     at void com.onesignal.OneSignalChromeTab.setup(android.content.Context, java.lang.String, java.lang.String, java.lang.String) (OneSignalChromeTab.java:46)
06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art:     at void com.onesignal.OneSignal$7.run() (OneSignal.java:1164)
06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art:     at void java.lang.Thread.run() (Thread.java:761)
06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.support.customtabs.CustomTabsServiceConnection" on path: DexPathList[[zip file "/data/app/br.com.jcs.torktransportes.torkapp-1/base.apk"],nativeLibraryDirectories=[/data/app/br.com.jcs.torktransportes.torkapp-1/lib/arm, /data/app/br.com.jcs.torktransportes.torkapp-1/base.apk!/lib/armeabi-v7a, /system/lib, /vendor/lib]]
06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art:     at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:56)
06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:380)
06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art:     at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art:     at void com.onesignal.OneSignalChromeTab.setup(android.content.Context, java.lang.String, java.lang.String, java.lang.String) (OneSignalChromeTab.java:46)
06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art:     at void com.onesignal.OneSignal$7.run() (OneSignal.java:1164)
06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art:     at void java.lang.Thread.run() (Thread.java:761)`

This didn't worked for me.

Stack trace:

06-30 09:11:23.334 29748-30190/br.com.jcs.torktransportes.torkapp E/AndroidRuntime: FATAL EXCEPTION: pool-7-thread-1 Process: br.com.jcs........, PID: 29748 java.lang.IllegalStateException: Default FirebaseApp is not initialized in this process br.com.jcs.torktransportes.torkapp. Make sure to call FirebaseApp.initializeApp(Context) first. at com.google.firebase.FirebaseApp.getInstance(Unknown Source) at com.google.firebase.iid.FirebaseInstanceId.getInstance(Unknown Source) at com.google.firebase.iid.FirebaseInstanceIdService.zza(Unknown Source) at com.google.firebase.iid.FirebaseInstanceIdService.handleIntent(Unknown Source) at com.google.firebase.iid.zzb$1.run(Unknown Source) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) at java.lang.Thread.run(Thread.java:761) 06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: Rejecting re-init on previously-failed class java.lang.Class<com.onesignal.OneSignalChromeTab$OneSignalCustomTabsServiceConnection>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/support/customtabs/CustomTabsServiceConnection; 06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: at void com.onesignal.OneSignalChromeTab.setup(android.content.Context, java.lang.String, java.lang.String, java.lang.String) (OneSignalChromeTab.java:46) 06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: at void com.onesignal.OneSignal$7.run() (OneSignal.java:1164) 06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: at void java.lang.Thread.run() (Thread.java:761) 06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.support.customtabs.CustomTabsServiceConnection" on path: DexPathList[[zip file "/data/app/br.com.jcs.torktransportes.torkapp-1/base.apk"],nativeLibraryDirectories=[/data/app/br.com.jcs.torktransportes.torkapp-1/lib/arm, /data/app/br.com.jcs.torktransportes.torkapp-1/base.apk!/lib/armeabi-v7a, /system/lib, /vendor/lib]] 06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:56) 06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:380) 06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312) 06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: at void com.onesignal.OneSignalChromeTab.setup(android.content.Context, java.lang.String, java.lang.String, java.lang.String) (OneSignalChromeTab.java:46) 06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: at void com.onesignal.OneSignal$7.run() (OneSignal.java:1164) 06-30 09:11:23.516 29748-30189/br.com.jcs.torktransportes.torkapp I/art: at void java.lang.Thread.run() (Thread.java:761)

--- a/OneSignalSDK/app/build.gradle
+++ b/OneSignalSDK/app/build.gradle
@@ -48,6 +48,7 @@ dependencies {
 //        exclude group: 'com.google.android.gms', module: 'play-services-gcm'
 //        exclude group: 'com.google.android.gms', module: 'play-services-analytics'
 //        exclude group: 'com.google.android.gms', module: 'play-services-location'
+//        exclude group: 'com.google.firebase', module: 'firebase-messaging'
     }
 
     // Use snapshot

--- a/OneSignalSDK/app/google-services.json
+++ b/OneSignalSDK/app/google-services.json
@@ -1,0 +1,42 @@
+{
+  "project_info": {
+    "project_number": "86233933250",
+    "firebase_url": "https://onesignalexample-8d956.firebaseio.com",
+    "project_id": "onesignalexample-8d956",
+    "storage_bucket": "onesignalexample-8d956.appspot.com"
+  },
+  "client": [
+    {
+      "client_info": {
+        "mobilesdk_app_id": "1:86233933250:android:37fb51944ca88b0f",
+        "android_client_info": {
+          "package_name": "com.onesignal.example"
+        }
+      },
+      "oauth_client": [
+        {
+          "client_id": "86233933250-qcj1t4q0lpon6q1nidojhrdh6gmgg3b6.apps.googleusercontent.com",
+          "client_type": 3
+        }
+      ],
+      "api_key": [
+        {
+          "current_key": "AIzaSyCvCs5vm7k3mo_2K0oB00QL7Z2zQKqSYY0"
+        }
+      ],
+      "services": {
+        "analytics_service": {
+          "status": 1
+        },
+        "appinvite_service": {
+          "status": 1,
+          "other_platform_oauth_client": []
+        },
+        "ads_service": {
+          "status": 2
+        }
+      }
+    }
+  ],
+  "configuration_version": "1"
+}

--- a/OneSignalSDK/build.gradle
+++ b/OneSignalSDK/build.gradle
@@ -13,6 +13,8 @@ buildscript {
     dependencies {
         classpath 'com.android.tools.build:gradle:3.1.0'
 
+        classpath 'com.google.gms:google-services:4.0.1'
+
         // OneSignal-Gradle-Plugin - Local testing
         // classpath 'com.onesignal:onesignal-gradle-plugin:[0.8.1, 0.99.99]'
 

--- a/OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java
+++ b/OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java
@@ -640,6 +640,8 @@ public class OneSignal {
 
       if (TrackFirebaseAnalytics.CanTrack())
          trackFirebaseAnalytics = new TrackFirebaseAnalytics(appContext);
+
+      PushRegistratorFCM.disableFirebaseInstanceIdService(appContext);
       
       initDone = true;
 

--- a/OneSignalSDK/onesignal/src/main/java/com/onesignal/PushRegistratorFCM.java
+++ b/OneSignalSDK/onesignal/src/main/java/com/onesignal/PushRegistratorFCM.java
@@ -27,18 +27,47 @@
 
 package com.onesignal;
 
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.pm.PackageManager;
+
 import com.google.firebase.FirebaseApp;
 import com.google.firebase.FirebaseOptions;
 import com.google.firebase.iid.FirebaseInstanceId;
+import com.google.firebase.iid.FirebaseInstanceIdService;
 import com.google.firebase.messaging.FirebaseMessaging;
 
 // TODO: 4.0.0 - Switch to using <action android:name="com.google.firebase.INSTANCE_ID_EVENT"/>
+// Note: Starting with Firebase Messaging 17.1.0 onNewToken in FirebaseMessagingService should be
+//   used instead.
 
 class PushRegistratorFCM extends PushRegistratorAbstractGoogle {
 
    private static final String FCM_APP_NAME = "ONESIGNAL_SDK_FCM_APP_NAME";
 
    private FirebaseApp firebaseApp;
+
+   // Disable in the case where there isn't a default Firebase app as this will crash the app.
+   //   The crash will happen where the Google Play services app fires an intent that the token
+   //      needs to be refreshed.
+   // This checks for gcm_defaultSenderId in values.xml (normally added from google-services.json)
+   // https://github.com/OneSignal/OneSignal-Android-SDK/issues/552
+   static void disableFirebaseInstanceIdService(Context context) {
+      String senderId = OSUtils.getResourceString(context, "gcm_defaultSenderId", null);
+      int componentState =
+         senderId == null ?
+         PackageManager.COMPONENT_ENABLED_STATE_DISABLED :
+         PackageManager.COMPONENT_ENABLED_STATE_ENABLED;
+
+      PackageManager pm = context.getPackageManager();
+      try {
+         ComponentName componentName = new ComponentName(context, FirebaseInstanceIdService.class);
+         pm.setComponentEnabledSetting(componentName, componentState, PackageManager.DONT_KILL_APP);
+      } catch (NoClassDefFoundError ignored) {
+         // Will throw if missing FirebaseInstanceIdService class, ignore in this case.
+         // We already print a logcat error in another spot
+      }
+   }
 
    @Override
    String getProviderName() {

